### PR TITLE
update for mithril 2

### PR DESCRIPTION
**Pull request by askvortsov1**

### CODE REVIEW

**Review by clarkwinkelmann — APPROVED** [review]

File: js/src/forum/index.js [hunk 1]
```diff
@@ -17,27 +17,25 @@ import MarkdownButton from './components/MarkdownButton';
...
 
   extend(TextEditor.prototype, 'view', function(vdom) {
     vdom.children[0].attrs.id = this.textareaId;
   });
 
-  extend(TextEditor.prototype, 'configTextarea', function(value, element, isInitialized, context) {
-    if (isInitialized) return;
-
-    const editor = new MarkdownArea(element, {
+  extend(TextEditor.prototype, 'textareaOncreate', function(value, vnode) {
```
Comment (franzliedke) [inline js/src/forum/index.js]: I am confused. This method (and the `Onremove` equivalent) don't seem to exist in core, in the `mithril-2-update` branch? :thinking:

Comment (clarkwinkelmann) [reply js/src/forum/index.js]: This was changed in https://github.com/flarum/core/pull/2255/commits/bedf019bdb96aefd2c3827feaaae114a85b90319

An update is necessary here

**Review by franzliedke — CHANGES_REQUESTED** [review]
Lifecycle hooks need to be fixed.

File: js/src/forum/index.js (hunk 1 above)
Comment (askvortsov1) [reply js/src/forum/index.js]: Fixed, tested locally.

File: js/src/forum/index.js [hunk 2]
```diff
@@ -17,27 +17,25 @@ import MarkdownButton from './components/MarkdownButton';
...
   extend(TextEditor.prototype, 'view', function(vdom) {
     vdom.children[0].attrs.id = this.textareaId;
   });
 
-  extend(TextEditor.prototype, 'configTextarea', function(value, element, isInitialized, context) {
-    if (isInitialized) return;
-
-    const editor = new MarkdownArea(element, {
+  extend(TextEditor.prototype, 'oncreate', function(value, vnode) {
+    this.editor = new MarkdownArea(vnode.dom, {
```
Comment (franzliedke) [inline js/src/forum/index.js]: Shouldn't this receive the `textarea` DOM node?

**Review by franzliedke — CHANGES_REQUESTED** [review]

**Review by franzliedke — APPROVED** [review]

**Pull request merged**